src/job_shop_scheduler_cqm.py

```diff
@@ -0,0 +1,237 @@
+    def __init__(self, model_data: JobShopData):
+        """
+        Args:
+            model_data: A JobShopData object that holds the data for this job shop 
+            scheduling problem.
+
+        """
+        self.model_data = model_data
+
+        self._process_data()
+        self._initialize_variables()
+        self.cqm = dimod.ConstrainedQuadraticModel()
+        self._add_task_start_constraint()
+        self._add_precedence_constraint()
+        self._add_share_machine_constraint()
+        self._remove_absurd_times()
+        # self._model_end()
+        self._add_objective()
+
+
+    def solve(self, profile: str=None) -> dict:
+        """Solves the job shop scheduling problem.
+        
+        Args:
+            profile: A string. The name of the profile to use for the hybrid solver.
+
+        Returns:
+            dict: A dictionary of the form {task: (start_time, end_time)} for each task in the
+            problem.
+        """
+        sampler = LeapHybridCQMSampler(profile=profile)
+        res = sampler.sample_cqm(self.cqm, time_limit=10, label='Job Shop Demo')
+
+        res.resolve()
+        feasible_sampleset = res.filter(lambda d: d.is_feasible)
+        try:
+            best_feasible = feasible_sampleset.first.sample
+
+        except ValueError:
+            raise RuntimeError(
+                "Sampleset is empty, try increasing time limit or " +
+                "adjusting problem config."
+            )
+
+        task_assignments = {}
+        for task in self.model_data.get_tasks():
+            for t in range(self.max_time):
+                if best_feasible[get_label(task, t)] == 1:
+                    start_time = t
+                    end_time = t + task.duration
+                    task_assignments[task] = (start_time, end_time)
+                    break
+
+        return task_assignments
+
+    
+    def _initialize_variables(self) -> None:
+        """Initialize the variables for the constraint satisfaction problem.
+        """
+        task_start_vars = {get_label(task, t): dimod.Binary(get_label(task, t)) 
+                        for t in range(self.max_time)
+                        for task in self.model_data.get_tasks()}
+        self.task_start_vars = task_start_vars
+
+
+    def _process_data(self):
+        """Process user input into a format that is more convenient for JobShopScheduler functions.
+        """
+        total_time = 0  # total time of all jobs
+        max_job_time = 0
+
+        for job in self.model_data.jobs:
+            job_time = self.model_data.get_total_job_time(job)
+            total_time += job_time
+
+            # Store the time of the longest running job
+            if job_time > max_job_time:
+                max_job_time = job_time
+
+        self.max_time = total_time
+
+
+    def _add_task_start_constraint(self) -> None:
+        '''
+        This adds this constraint to the model requiring that each task be started exactly once.
+        '''
```

Comment:
```suggestion
        """
        This adds this constraint to the model requiring that each task be started exactly once.
        """
```
nit here and elsewhere - comment consistency